### PR TITLE
add integration test script runner

### PR DESCRIPTION
**Pull request by linuxwolf**
Adds "integration" to the package.json scripts, in order to run `tox`-based integration tests, with prerequisites.

### CODE REVIEW

**Comment by linuxwolf** [comment]
There is no hurry on getting this landed.  Merely a convenience for those of us working with integration tests a lot.

**Comment by jrbenny35** [comment]
We should have a ```flake8``` as well as an ```integration test``` step. It would help with CI, to not have such a huge log if flake8 fails.

**Comment by jimporter** [comment]
@jrbenny35 Would it make sense to merge the `flake8` and `eslint` checks into a single omni-lint step? (Plus `stylelint` one day.) I don't know if we need to break those out into separate things.

**Comment by jrbenny35** [comment]
> Would it make sense to merge the flake8 and eslint checks into a single omni-lint step? (Plus stylelint one day.) I don't know if we need to break those out into separate things.

@jimporter I like it, if that would help. Just lint ALL THE THINGS.

**Comment by devinreams** [comment]
>Would it make sense to merge the flake8 and eslint checks into a single omni-lint step?

@jimporter And if we do that, we take it out as a stage in the travis.yml right? Because it would run as part of test, if I understand correctly?

File: package.json [hunk 1]
```diff
@@ -36,6 +36,8 @@
     "run": "jpm run --addon-dir dist/ ${npm_config_jpm_runflags} --prefs jpm-prefs.json",
     "pretest": "npm run-script autolint",
     "test": "cross-env NODE_ENV=test karma start",
+    "preintegration": "cross-env ENABLE_DOORHANGER=1 npm run package",
```
Comment (pdehaan) [inline package.json]: Just as a sanity check, do we still need this ENV var magic?
I thought we yanked all the `ENABLE_DOORHANGER` in PR #403; "Remove ENABLE_DOORHANGER code".

**Comment by linuxwolf** [comment]
@jimporter I thought about changing around `test`, but figured that might be too much for this first go around.

File: package.json (hunk 1 above)
Comment (linuxwolf) [reply package.json]: will remove; when first submitted this was still necessary.

**Comment by linuxwolf** [comment]
rebased onto `master` and made the requested changes.

**Pull request merged**